### PR TITLE
Enable nullability in some ControlDesigner members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some ControlDesigner members.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11285)

### CODE REVIEW

**Comment by SimonZhao888** [comment]
LGTM!

**Review by SimonZhao888 — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 1]
```diff
@@ -927,14 +932,14 @@ public override void Initialize(IComponent component)
...
-        DockingAttribute dockingAttribute = (DockingAttribute)attributes[typeof(DockingAttribute)];
+        DockingAttribute? dockingAttribute = (DockingAttribute?)attributes[typeof(DockingAttribute)];
         if (dockingAttribute is not null && dockingAttribute.DockingBehavior != DockingBehavior.Never)
         {
             // Create the action for this control
             _dockingAction = new DockingActionList(this);
 
             // Add our 'dock in parent' or 'undock in parent' action
-            if (TryGetService(out DesignerActionService das))
+            if (TryGetService(out DesignerActionService? das))
```
Comment (Epica3055) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:942]:
```suggestion
            if (TryGetService(out DesignerActionService? designerActionService))
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 2]
```diff
@@ -1029,15 +1036,15 @@ private void OnControlRemoved(object sender, ControlEventArgs e)
...
+    private void DataBindingsCollectionChanged(object? sender, CollectionChangeEventArgs e)
     {
         // It is possible to use the control designer with NON CONTROl types.
         if (Component is Control ctl)
         {
             if (ctl.DataBindings.Count == 0 && _removalNotificationHooked)
             {
                 // Remove the notification for the ComponentRemoved event
-                if (TryGetService(out IComponentChangeService csc))
+                if (TryGetService(out IComponentChangeService? csc))
```
Comment (Epica3055) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1047]:
```suggestion
                if (TryGetService(out IComponentChangeService? componentChangeService))
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 3]
```diff
@@ -1047,7 +1054,7 @@ private void DataBindingsCollectionChanged(object sender, CollectionChangeEventA
             else if (ctl.DataBindings.Count > 0 && !_removalNotificationHooked)
             {
                 // Add the notification for the ComponentRemoved event
-                if (TryGetService(out IComponentChangeService csc))
+                if (TryGetService(out IComponentChangeService? csc))
```
Comment (Epica3055) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1057]:
```suggestion
                if (TryGetService(out IComponentChangeService? componentChangeService))
```

**Review by Epica3055 — COMMENTED** [review]
please avoid using abbreviations

**Review by ricardobossan — COMMENTED** [review]
LGTM

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 4]
```diff
@@ -1110,7 +1117,7 @@ public override void InitializeExistingComponent(IDictionary defaultValues)
         {
             if (c is not null)
             {
-                ISite site = c.Site;
+                ISite? site = c.Site;
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1120]:
```suggestion
                ISite? site = control.Site;
```

**Review by LeafShi1 — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**